dids/src/main/kotlin/web5/sdk/dids/methods/web/DidWeb.kt

```diff
@@ -192,15 +192,15 @@ public sealed class DidWebApi(
   private fun decodeId(parsedDid: Did): String {
```

Comment:
@aparkersquare thoughts on using a parsed url to build `targetUrl`? something like

```kotlin
  public fun decodeId(parsedDid: Did): String {
    val domainNameWithPath = parsedDid.id.replace(":", "/")
    val decodedDomain = URLDecoder.decode(domainNameWithPath, UTF_8)
    val url = URL("https://$decodedDomain")

    val decodedId = StringBuilder()
    if (url.host == "localhost" || url.host == "127.0.0.1") {
      decodedId.append("http://")
    } else {
      decodedId.append("https://")
    }

    decodedId.append(decodedDomain)

    if (url.path.isEmpty()) {
      decodedId.append(WELL_KNOWN_URL_PATH)
    }
    decodedId.append(DID_DOC_FILE_NAME)

    return decodedDomain.toString()
  }
```

potential benefits of doing it this way are:
* url is parsed (therefore "validated" prior to doing anything else
* we can check `url.host` explicitly

there's probably kotlin idioms that can be used in the above suggestion to make the code more kotlin-esque which you'd know better about than me!

when i gave chad (aka chatGPT) the above suggestion and asked him for something more idiomatic, he suggested:

```kotlin
public fun decodeId(parsedDid: Did): String {
    val domainNameWithPath = parsedDid.id.replace(":", "/")
    val decodedDomain = URLDecoder.decode(domainNameWithPath, UTF_8.name())
    val url = URL("https://$decodedDomain")

    return buildString {
        append(if (url.host == "localhost" || url.host == "127.0.0.1") "http://" else "https://")
        append(decodedDomain)
        if (url.path.isEmpty()) {
            append(WELL_KNOWN_URL_PATH)
        }
        append(DID_DOC_FILE_NAME)
    }
}
```

let me know what you think! i personally prefer the former but maybe that's because i don't write much kotlin. Either way, 👍🏽

Reply:
Oooh, I didn't know about `url.host`!
Agree, that looks like it could be a nicer approach. Thanks.

I'll have a closer look soon.

Reply:
This works.
Updated the PR to use this approach, and deleted the LocalHostDomainMatcher

Reply:
Naisuu!